### ISSUE TITLE
pvsbin artifacts

### ISSUE THREAD

**Issue by tjingboem** [csound/csound#1530]
With pvsbin, why are there artifacts produced?

```
<CsoundSynthesizer>
<CsOptions>
; Select audio/midi flags here according to platform
-odac      ;;;realtime audio out
;-iadc    ;;;uncomment -iadc if realtime audio input is needed too
; For Non-realtime ouput leave only the line below:
; -o pvsbin.wav -W ;;; for file output any platform
</CsOptions>
<CsInstruments>

; by Richard Boulanger & Menno Knevel

sr = 44100
ksmps = 32
nchnls = 2
0dbfs  = 1

instr 1

ifftsize = 1024
iwtype   = 1                                      
a1 diskin2 "fox.wav", 1, 0, 1
fsig pvsanal a1, ifftsize, ifftsize/4, ifftsize, iwtype
kamp, kfr pvsbin fsig, p4
adm poscil kamp, kfr
outs adm, adm

endin

</CsInstruments>
<CsScore>
;           bin
i 1 0 2.7   5
i 1 3 .     10
i 1 6 .     15
i 1 9 .     20

i 1 13 2.7  5   ; chord
i 1 13 .    10
i 1 13 .    15
i 1 13 .    20
e
</CsScore>
</CsoundSynthesizer>
```

**Comment by vlazzarini** [comment]
because picking up a single bin from an pv frame is prone to artefacts, I would expect that.

**Comment by tjingboem** [comment]
Ah, thanks.

In the example i read: /* cleaner with hanning window */
pvsanal does not provide the Hanning window, so i can't check that.
pvsanal has: 0 = Hamming window, 1 = von Hann window, 3 = Kaiser window (not in sliding form)
Is it worth the trouble to want to check and see if these artifacts would indeed be less, you think?

**Comment by vlazzarini** [comment]
von Hann and Hanning are the same thing. Hanning is cleaner for everything so that is the standard for all cases.

**Comment by vlazzarini** [comment]
you can try a smaller hopsize. Also using port on freq and/or amp may smooth it if there spikes.

**Comment by tjingboem** [comment]
I checked Wikipedia for this, and although this not being a scientific approach, i read:
"...named after Julius von Hann, and sometimes referred to as Hanning, presumably due to its linguistic and formulaic similarities to the Hamming window. "
I did not know that this is in fact the same window :-)

I tried reducing the hopsize, and it does sound a little better in the sense that the artifacts now become more a part of the sound instead of a bunch of small separate clicks. 
I'll add this remark to the example: 
; to reduce artifacts, try a smaller hopsize, like ifftsize/8
; also using port on frequency and/or amplitude may smooth it if there spikes.

Thanks for your comments!
